bin/cheritest/cheritest_syscall.c

```diff
@@ -232,3 +234,54 @@ test_ptrace_basic(const struct cheri_test *ctp __unused)
 		exit(23);
 	}
 }
+
+static void
+test_aio_sival_handler(int sig __unused, siginfo_t *si, void *uc __unused)
+{
+	CHERITEST_VERIFY2(si->si_code == SI_ASYNCIO,
+		"Signal not asyncio?  code=%d\n", si->si_code);
+	CHERITEST_VERIFY2(si->si_value.sival_ptr == test_aio_sival_handler,
+		"Bad si_value; expected=%p got=%p\n",
+		test_aio_sival_handler, si->si_value.sival_ptr);
+	cheritest_success();
+}
+
+void
+test_aio_sival(const struct cheri_test *cpt __unused)
+{
+	char buf[128];
+	int pfd[2];
+	int res;
+	sigset_t sigset, osigset;
+	struct aiocb aiocb;
+	struct sigaction sa;
+
+	sa.sa_flags = SA_SIGINFO;
+	sa.sa_sigaction = &test_aio_sival_handler;
+	res = sigaction(SIGUSR1, &sa, NULL);
+	CHERITEST_VERIFY2(res == 0, "Could not install signal handler; errno=%d", errno);
+
+	res = socketpair(AF_UNIX, SOCK_STREAM, 0, pfd);
+	CHERITEST_VERIFY2(res == 0, "Could not create socketpair; errno=%d", errno);
+
+	aiocb.aio_fildes = pfd[0];
+	aiocb.aio_buf = buf;
+	aiocb.aio_nbytes = sizeof buf;
+	aiocb.aio_lio_opcode = LIO_READ;
+
+	aiocb.aio_sigevent.sigev_notify = SIGEV_SIGNAL;
+	aiocb.aio_sigevent.sigev_signo = SIGUSR1;
+	aiocb.aio_sigevent.sigev_value.sival_ptr = test_aio_sival_handler;
+
+	res = aio_read(&aiocb);
+	CHERITEST_VERIFY2(res == 0, "Could not register aio; errno=%d", errno);
+
+	sigemptyset(&sigset);
+	sigaddset(&sigset, SIGUSR1);
+	sigprocmask(SIG_BLOCK, &sigset, &osigset);
+	close(pfd[1]);
+	sigsuspend(&osigset);
```

Comment:
Please use `alarm` or something to get a timeout, and this should probably be checking aio_error is 0 too (after shifting the other checks out of the signal handler).